programs/manifest/src/state/resting_order.rs

```diff
@@ -223,11 +223,18 @@ impl PartialOrd for RestingOrder {
 
 impl PartialEq for RestingOrder {
     fn eq(&self, other: &Self) -> bool {
-        // Only used in equality check of lookups. To enable coalescing, just
-        // check price, trader, and order type.
-        (self.price == other.price)
-            && (self.trader_index == other.trader_index)
-            && (self.order_type == other.order_type)
+        if self.trader_index != other.trader_index || self.order_type != other.order_type {
+            return false;
+        }
+        if self.order_type == OrderType::Reverse {
+            // Allow off by 1 for reverse orders to enable coalescing. Otherwise there is a back and forth that fragments into many orders.
```

Comment:
Should we enforce this to only make the price worse? Some edge case where that atom improvement tries to make it match?

Reply:
This is 10^-18 and only when there is an existing order already to match. We should not move an existing order.
This isnt order matching or sorting, just lookup for coalescing into an existing reverse order

Reply:
Should we be able rule out one of the cases? +1 or -1 not both?

Reply:
Maybe it only gets called one way in practice because of coincidental details of how this is called and only one direction has the error. However it is preferable to keep an equality function to have commutativity.

Reply:
Seems fine might be worth considering if sell then equal only if price is -1 and only +1 for buys